chore: Adds linting + clean & nuke scripts

grebaldi opened the pull request:
This PR adds linting via `editorconfig-checker` and `prettier`.

Furthermore, it externalizes all package.json scripts to a `scripts/` directory and adds a `clean` and a `nuke` script.

grebaldi committed:
chore: Adds linting via `prettier` and `editorconfig-checker`

grebaldi committed:
chore: Externalizes all scripts

grebaldi committed:
chore: Adds `clean` script

grebaldi committed:
chore: Adds `nuke` script

henrikradandt reviewed (approved):
All scripts, except pre-commit-hook, work as expected.

grebaldi committed:
chore: Adds postinstall script that sets up husky

grebaldi committed:
chore: Adds pre-commit hook

grebaldi committed:
chore(use-business): Fixes code style

grebaldi commented:
Hi @henrikradandt,

thanks for testing this so thoroughly :+1: 

This helped me uncover a major API change in `husky` that totally slipped me. (see: https://typicode.github.io/husky/#/?id=husky-4-to-7-cli)

I fixed the setup and added a postinstall script, so that the git hooks are being properly installed right after running `pnpm install`.

I also rebased this branch on `main` and applied the new linting to the recently added `use-business` package.

henrikradandt reviewed (approved):
Hi @grebaldi, nice. Now linting works as expected on pre-commit. I successfully have tested that only staged changes will be linted and also prettier write mode for auto applyable fixes does its job.
Also use-business is integrated in this script setup.

grebaldi committed:
chore(use-pagination): Fixes code style

grebaldi committed:
chore: Fixes code style in README.md
